sdk/servicebus/azure-servicebus/azure/servicebus/_pyamqp/aio/_transport_async.py

```diff
@@ -575,6 +600,8 @@ async def _read(self, toread, buffer=None, **kwargs):  # pylint: disable=unused-
                     self._read_buffer = BytesIO(data[toread:])
                     toread = 0
             return view
+        except AttributeError:
```

Comment:
@annatisch  TL;DR - when pickling `self.ws.receive_bytes()` raises an AttributeError b/c self.ws = None. Do we want the try except to be on line 592 at the lowest level possible or here? (In sync, it's here outside the while loop.)

when pickling + pickle.loading a receiver object + underlying transport, we can't store/re-set the self.ws object in the __setstate__ method b/c the underlying connection is closed by remote host. So it needs to be re-created. In order to do this, it must first raise an IOError here, so the connection will know the underlying socket has been closed and will re-create in the retry loop. `self.ws.receive_bytes()` will raise an AttributeError to indicate that self.ws = None. In sync, we catch the exception here. Should we do the same here? Or just wrap around line 592?